### PR TITLE
Add extra arguments to hubert pretrain factory functions

### PR DESCRIPTION
**Pull request by nateanl**
In different pre-training and fine-tuning settings, the `mask_prob`, `mask_channel_prob`, and `mask_channel_length` are different. For example, the settings in [pre-training](https://github.com/pytorch/fairseq/blob/main/examples/hubert/config/pretrain/hubert_base_librispeech.yaml#L70) and [fine-tuning](https://github.com/pytorch/fairseq/blob/main/examples/hubert/config/finetune/base_10h.yaml#L69-L73) are different. The motivation is to avoid overfitting when fine-tuning on a small dataset (example: [fine-tune on 10 minutes of audio](https://github.com/pytorch/fairseq/blob/main/examples/wav2vec/config/finetuning/vox_10m.yaml#L57-L59)).
This PR adds the required arguments in the factory functions to make them tunable for pre-training and fine-tuning. `mask_length` is set to `10` by default for all cases, hence it's not included in the factory function.

### CODE REVIEW

File: torchaudio/models/wav2vec2/model.py [hunk 1]
```diff
@@ -1033,6 +1042,9 @@ def hubert_pretrain_large(
     encoder_ff_interm_dropout: float = 0.0,
     encoder_dropout: float = 0.0,
     encoder_layer_drop: float = 0.0,
+    mask_prob: float = 0.8,
+    mask_channel_prob: float = 0.0,
+    mask_channel_length: float = 10,
 ) -> HuBERTPretrainModel:
     # Overriding the signature so that the return type is correct on Sphinx
     """hubert_pretrain_large(encoder_projection_dropout: float = 0.0, encoder_attention_dropout: float = 0.0, encoder_ff_interm_dropout: float = 0.0, encoder_dropout: float = 0.0, encoder_layer_drop: float = 0.0) -> torchaudio.models.HuBERTPretrainModel
```
Comment (carolineechen) [inline torchaudio/models/wav2vec2/model.py:1050]: remember to add the new mask params to the signature override as well

File: torchaudio/models/wav2vec2/model.py [hunk 2]
```diff
@@ -1096,6 +1114,9 @@ def hubert_pretrain_xlarge(
     encoder_ff_interm_dropout: float = 0.0,
     encoder_dropout: float = 0.0,
     encoder_layer_drop: float = 0.0,
+    mask_prob: float = 0.8,
+    mask_channel_prob: float = 0.0,
+    mask_channel_length: float = 10,
 ) -> HuBERTPretrainModel:
     # Overriding the signature so that the return type is correct on Sphinx
     """hubert_pretrain_xlarge(encoder_projection_dropout: float = 0.0, encoder_attention_dropout: float = 0.0, encoder_ff_interm_dropout: float = 0.0, encoder_dropout: float = 0.0, encoder_layer_drop: float = 0.0) -> torchaudio.models.HuBERTPretrainModel
```
Comment (carolineechen) [inline torchaudio/models/wav2vec2/model.py:1122]: same as above -- add new params to sphinx signature override

File: torchaudio/models/wav2vec2/model.py [hunk 3]
```diff
@@ -967,6 +967,9 @@ def hubert_pretrain_base(
     encoder_ff_interm_dropout: float = 0.0,
     encoder_dropout: float = 0.1,
     encoder_layer_drop: float = 0.05,
+    mask_prob: float = 0.8,
+    mask_channel_prob: float = 0.0,
+    mask_channel_length: float = 10,
     num_classes: int = 100,
 ) -> HuBERTPretrainModel:
     # Overriding the signature so that the return type is correct on Sphinx
```
Comment (carolineechen) [inline torchaudio/models/wav2vec2/model.py:975]: here as well

**Review by carolineechen — COMMENTED** [review]
the factory functions here support the `mask_channel_length` parameter unlike what the PR summary describes -- which one should it be?

**Comment by nateanl** [comment]
`mask_channel_length` is the one that is included. Since `mask_length` value doesn't change in pre-training and fine-tuning, it will not be included in the argument list.

**Review by carolineechen — APPROVED** [review]
Looks good, thanks
